### ISSUE TITLE
App not pairing (issue 8 not solved)

### ISSUE THREAD

**Issue by db69** [RfidResearchGroup/ChameleonMini#12]
Flashed the latest versions from repo as indicated.

Chameleon version? output : 
"ChameleonMini RevG 191226 using LUFA 151115 compiled with AVR-GCC 4.9.2. Based on the open-source NFC tool ChameleonMini. https://github.com/emsec/ChameleonMini commit d6d183b"

Running on Oneplus 6T

The app does seem to detect the device, but answers "battery low or incompatible firmware"

**Comment by maxieds** [comment]
Yes, confirmed as still an issue! I am having the same issue on a new Motorola phone using Android 9. An image is attached. 

<img src="https://user-images.githubusercontent.com/22165688/77207075-d81fee80-6ace-11ea-91c3-69add5dbfae4.png" width="250" />

I am also trying to get bluetooth support working within my Android app, the [Chameleon Mini Live Debugger](https://github.com/maxieds/ChameleonMiniLiveDebugger). So far [my attempts](https://github.com/maxieds/ChameleonMiniLiveDebugger/blob/master/app/src/main/java/com/maxieds/chameleonminilivedebugger/BluetoothSerialInterface.java) at bluetooth connections to the device are unsuccessful. The Droid and my PC do however recognize the device as named "**BLE-Chameleon**". The serial USB exchanges with my application work as expected with the latest firmware.

**Comment by maxieds** [comment]
Since this issue still remains unresolved, I'm going to ask the obvious question that occurred to me testing with the proxgrind boards again: Is it possible that some of these devices shipped to production with completely broken BlueTooth implementations? My reasoning is this: 
1. The docs say you have to use the closed-source Android app that ships from the proxgrind developers sources to re-flash the BT firmware; 
2. Said app does not detect the devices well enough yet to pair with the BT chip that comes stock on a fresh device. 

Therefore, I have to ask how exactly the BT gets updated to working status at this point? 

I am not complaining too much because I received free testing devices to work with the proxgrind devices. They are also significantly sturdier than the KAOS variants and harder to break (a very nice feature). But, I still need to know how to get BT working on one of these things with a broken Android implementation. 

Any suggestions are of course welcome.

**Comment by db69** [comment]
In my case, the issue is not related to broken hardware : the Chameleon 
pairs nicely (although uselessly :-) ) with other BT devices like my 
laptop.  Furthermore, it is detected by the app. It is only at pairing 
time that things go wrong

Le 16-04-20 à 19:45, Maxie D. Schmidt a écrit :

**Comment by db69** [comment]
Finally worked by uninstalling chameleon from android, installing  
android apk from the repository (not Google or others), re-installing 
atmel drive and flashing firmware again. All files taken from 
RRRG/Chameleonmini repo, Proxgrind branch 
https://github.com/RfidResearchGroup/ChameleonMini/blob/proxgrind/REV.G%20User%20Manual%2020200309.zip 
needs unzipping)

Le 16-04-20 à 21:49, db69b a écrit :

**Comment by 4NTC4RD** [comment]
Hi. Make sure that when you connect the bluetooth to access the chameleon app you also have to have the location on the phone activated. Press a button on the chameleon, click on the app and ble-chameleon will appear. you hit the connect button on the app and you're done. I hope it helps. Sorry for my English. Greetings

**Comment by maichai** [comment]
I bought my Mini at Sneak.technology and it was delivered with this firmware: ChameleonMini RevG compiled at 2020-02-03 17:22:16(11a8cc5) using LUFA 151115 with AVR-GCC 5.4.0.

I can connect to it using the laptop via USB and use the terminal.

However, the Mini never shows in any BT scan of any device (laptop, android phone). On the phone locations is also enabled.

Is my Mini broken? Is the firmware bad?

**Comment by FallingLights** [comment]
What Android version are you running?
I checked the app on 3 Phones Android: 7,9,10. And only on Android 7, the app was able to detect the Chameleon and connect to it.
So I'm guessing... the APP is broken

**Comment by shellchocolat** [comment]
not working for me anymore.

I upgraded the firmware on the Chameleon.

I downloaded the app on the playstore, but i also tried with one available on the repo.

I am pretty confident that the bluetooth on the chameleon is working as i can sniff its traffic with my Ubertooth, and i saw some broadcast messages (ADV_IND) from the Chameleon. But the app doesn't seems to respond.

More, i tried to look into the apk, but the code seems definitely obfuscated ....

I used on Motoral Edge device with Android v10 on it.

Maybe an update of the app could be great

**Comment by maichai** [comment]
> What Android version are you running?
> I checked the app on 3 Phones Android: 7,9,10. And only on Android 7, the app was able to detect the Chameleon and connect to it.
> So I'm guessing... the APP is broken

In my case I cannot follow this logic: none of my devices sees the ChameleonMini at all (macOS, Kali bettercap, Android, IOS) - it is not only the Chameleon app that is not seeing it (device does not show up, so it cannot be selected. It also does not show up in the RRG RFID Tools App (although there the Chameleon is labeled RevE, and I have a RevG)

**Comment by karmathub** [comment]
Hi there,
It's need to activare geolocalization on latest Android releases, it works fine.
